GCTC-NTGC/gc-digital-talent
# Use fully-formatted Job Poster Title on ViewPool and ViewUser admin pages

Resolves #4684.
Resolves #4680.

~~*Note*: This overlaps a lot with #4599. Whichever is merged second should make sure they make use of the same title-formatting function. This PR simply uses the function on two more admin pages.~~

This PR replaces nearly every use of `pool.name` with a call to `getFullPoolAdvertisementTitle`. The only exception is on the Edit Pool Advertisement page, where you edit the `pool.name` field in an input element.

Additionally, this means that any graphql operation which was retrieving a pool with name included now also requests classification group+level, and stream. Any operations which requested only id Ieft untouched. Note that in any future places where we request a pool and want to display its name, we must request name, classifications, and stream at the same time.

## Testing
1. Run database seeder: `docker-compose exec -w /var/www/html/api php sh -c "php artisan migrate --seed"`
2. Log in as an admin and go to http://localhost:8000/en/admin/pools
3. Find a pool which includes a classification and stream in its name, and click the name to go to the ViewPool page
4. Confirm that the classification and stream appear in the title at the top of the page
5. Click `Manage Candidates`
6. Find a candidate in the table whose action link says "View Profile" instead of "View Application". If they all say "View Application", view the application, change the status to Placed Casual, and return.
7. Click View Profile on a candidate
8. Confirm that the user's table of Pools shows the original pool name with classification and stream included.
9. Confirm that you can click the pool name to return to the ViewPool page.
10. Find a Draft pool and click Edit.
11. Confirm that the Specific Title fields (in english and in french) do NOT include classification or stream.
12. Ensure the draft pool has a classification and a stream, and has a closing date in the future, and is in the IT Jobs publishing group. Publish it.
13. Go to the Browse Opportunities page (http://localhost:8000/en/browse/pools)
14. Confirm that your new PoolAdvertisement appears as a card with the fully formatted title.
15. Click apply on the card.
16. Confirm the Pool page shows the fully formatted title in the hero and in the breadcrumbs.
